superset/charts/data/api.py

```diff
@@ -394,8 +394,13 @@ def _process_data(query_data: Any) -> Any:
             )
 
         if result_format == ChartDataResultFormat.JSON:
+            queries = result["queries"]
+            if security_manager.is_guest_user():
+                for query in queries:
+                    with contextlib.suppress(KeyError):
+                        del query["query"]
```

Comment:
This works too, `contextlib.supress` seems like using a canon to kill a fly. :)

```suggestion
                    query.pop("query", None)
```